src/compiler/checker.ts

```diff
@@ -1834,22 +1831,47 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         typeHasCallOrConstructSignatures,
     };
 
-    function runWithoutResolvedSignatureCaching<T>(node: Node | undefined, fn: () => T): T {
-        const cachedSignatures = [];
-        while (node) {
-            if (isCallLikeExpression(node)) {
-                const nodeLinks = getNodeLinks(node);
-                const resolvedSignature = nodeLinks.resolvedSignature;
-                cachedSignatures.push([nodeLinks, resolvedSignature] as const);
-                nodeLinks.resolvedSignature = undefined;
+    function getCandidateSignaturesForStringLiteralCompletions(call: CallLikeExpression, editingArgument: Node, checkMode = CheckMode.IsForStringLiteralArgumentCompletions) {

```

Comment:
This can be simplified a little because now this function is capable of doing the "double request" for completions but that has been conditionally already done in `stringCompletions.ts`. So now we can simplify the code in there and rely solely on what you introduced here.

To illustrate what I exactly mean I prepared a small PR targeting this one: https://github.com/microsoft/TypeScript/pull/55624